rai/lib/blocks.cpp

```diff
@@ -1440,6 +1460,11 @@ rai::block_hash rai::receive_block::root () const
 	return hashables.previous;
 }
 
+rai::block_hash rai::receive_block::link () const
+{
+	return 0;
```

Comment:
Should we `assert(false)` here (or just not implement this virtual function for non-state blocks) ?  The receive block probably has the equivalent of a link

Reply:
That's similar to source (), previous () & some other functions returning 0 for some block types. We can just not implement such functions when they return 0 I guess to save some lines in code

Reply:
Is this a case of using release_assert since we shouldn’t be looking for the link of non-state blocks?

Reply:
Either release_assert() or not declaring the virtual methods for these block types which should generate a fatal exception